hoomd/update/__init__.py

```diff
@@ -0,0 +1,5 @@
+from hoomd.update.box_resize import BoxResize
+
+
+class _updater:
+    pass
```

Comment:
Can do.

Reply:
This comment was resolved but no action was taken?

Reply:
Sorry, I must have accidentally closed it. I try to implement most of the changes by going one by one. I don't know how I resolved this without implementing the change.

Reply:
@b-butler You added it and I removed it. When I build the sphinx documentation with your `__all__`, I get a warning: 
```
WARNING: __all__ should be a list of strings, not [<class 'hoomd.update.box_resize.BoxResize'>] (in module hoomd.update) -- ignoring __all__
```

Also, we do not use `__all__` anywhere else in HOOMD, nor do we encourage or support `import *` usage. What purpose does adding `__all__` here serve?

Reply:
That is because, I used `__all__` wrong. I passed it the class itself (forgot to quote it).

Sphinx respects auto documenting things found in `__all__` in the order they are specified. Otherwise, not much besides the `from a import *` syntax which I am not a fan of.

Reply:
We recently had a discussion about the use of `__all__` during the groups PR in signac-flow, [that discussion]
(https://github.com/glotzerlab/signac-flow/pull/114#issuecomment-586416971) might be helpful to consider here.

Reply:
It is still true that `__all__` is used nowhere else in HOOMD-blue. It is out of the scope of this PR to change that. Feel free to raise another issue or PR. While PEP8 encourages both `__all__` **and** initial underscores for private names, note that `pycodestyle` does not issue a warning or error for missing `__all__` variables. 

Changing this behavior across all of HOOMD would require some other tool to style check that `__all__` is specified and a wealth of additional tests to ensure that all classes that should be added to `__all__` are added. I would much prefer to have a single source of truth for public/private APIs as this requires no additional testing.

Reply:
I agree that this issue is out of scope for this PR. I'm happy to have a broader discussion on this at a later date. I think it's a discussion that should be had so that we can reach consensus, but I don't think it's a priority until the API is more settled anyway.

Reply:
I will `__all__` again, so we can decide on this later and make all the changes (if we decide to) simultaneously.